From f58ebc7e755e7baa122b906e061feb4de10bbe4c Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Fri, 14 Dec 2018 17:02:11 +0000
Subject: [PATCH 6/8] pc: Add x-migrate-smi-count=off to PC_RHEL7_6_COMPAT

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20181214170211.14267-7-dgilbert@redhat.com>
Patchwork-id: 83514
O-Subject: [RHEL8 qemu-kvm PATCH v3 6/6] pc: Add x-migrate-smi-count=off to PC_RHEL7_6_COMPAT
Bugzilla: 1655820
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

MSR_SMI_COUNT started being migrated in QEMU 2.12 and in the 2.12
release this broke back migration to earlier versions;  however
that didn't cause a problem on RHEL because it also relied on newer
kernel features that RHEL 7.* doesn't have.

QEMU 3.0 got a fix (in PC_COMPAT_2_11) to fix the 2.12->earlier
breakage, but given the kernel dependency, it makes more sense
for us to tie it to 8.* machine types and keep the feature off for
all 7.* machine types.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 include/hw/i386/pc.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/hw/i386/pc.h b/include/hw/i386/pc.h
index 5533863..426a975 100644
--- a/include/hw/i386/pc.h
+++ b/include/hw/i386/pc.h
@@ -1000,6 +1000,9 @@ extern void igd_passthrough_isa_bridge_create(PCIBus *bus, uint16_t gpu_dev_id);
 
 /* Similar to PC_COMPAT_3_0 + PC_COMPAT_2_12, but:
  * all of the 2_12 stuff was already in 7.6 from bz 1481253
+ * x-migrate-smi-count comes from PC_COMPAT_2_11 but
+ * is really tied to kernel version so keep it off on 7.x
+ * machine types irrespective of host.
  */
 #define PC_RHEL7_6_COMPAT \
         HW_COMPAT_RHEL7_6 \
@@ -1015,6 +1018,10 @@ extern void igd_passthrough_isa_bridge_create(PCIBus *bus, uint16_t gpu_dev_id);
             .driver   = "Skylake-Server-IBRS" "-" TYPE_X86_CPU,\
             .property = "pku",\
             .value    = "off",\
+        },{ /* PC_RHEL7_6_COMPAT from PC_COMPAT_2_11 */ \
+            .driver   = TYPE_X86_CPU,\
+            .property = "x-migrate-smi-count",\
+            .value    = "off",\
         },
 
 /* Similar to PC_COMPAT_2_11 + PC_COMPAT_2_10, but:
-- 
1.8.3.1

